lib/handlers/allow.js

```diff
@@ -23,6 +24,11 @@ function allow (mode) {
       ? res.locals.path
       : req.path
 
+    // If a relativePath has been provided, check permissions based on that
+    if (testDirectory) {
+      reqPath = path.dirname(reqPath)
```

Comment:
Maybe deceptive to use `reqPath` for it then. Perhaps should be something like `resourcePath`.